src/layer_builder.rs

```diff
@@ -151,13 +154,13 @@ where
 
         let layer_bundle = LayerBundle {
             layer,
-            transform: Transform::from_xyz(0.0, 0.0, settings.layer_id as f32),
+            transform: Transform::from_xyz(0.0, 0.0, layer_id as f32),
             ..LayerBundle::default()
         };
 
-        let mut layer = layer_bundle.layer;
+        let layer = layer_bundle.layer;
         let mut transform = layer_bundle.transform;
-        layer.settings.layer_id = layer.settings.layer_id;
+        // TODO: is this correct?
         transform.translation.z = 0.0; //layer.settings.layer_id as f32;
```

Comment:
This isn't correct. I think the solution in #139 is a bit better in that regard.